ホームページでは自身の分野の投稿のみを表示するように

takumi-333 opened the pull request:
- `(home)/page.tsx`でcurrentUserからfieldsを受け取るようにしました。
ユーザーがなければnullを渡す
- `Reviews.tsx`では受け取ったfieldsとtagをもとにfetchReviewsByFields()します 
- `utils.ts`で特定のstringが特定のstring[]配列に含まれるかをチェックする
checkInStringArray()関数を定義しました。

takumi-333 committed:
[update]
・ホームページで自身の分野の投稿のみを表示するようにしました。
・(home)/page.tsxでcurrentUserからfieldsを受け取るようにしました。
なければnullを渡す
・Reviews.tsxでは受け取ったfieldsとtagをもとにfetchReviewsByFields()します
・utils.tsで特定のstringが特定のstring[]配列に含まれるかをチェックする
checkInStringArray()関数を定義しました。

DoiRyoto commented on app/(home)/page.tsx:
### 16行目

let使うと思わぬバグが入るかもなので、極力letは使わない方が良いと思う
```const userInfo = _user ? await fetchUser(_user.id) : null```で良いんでは？

三項演算子使わないなら、letで先にnullにする(_userが存在しない場合はnullにするという意図を、16行目のコードに持たせる)という書き方は直感的じゃないので、きちんと条件分岐を書いた方が良いと思う。

DoiRyoto replied:
https://typescriptbook.jp/reference/values-types-variables/let-and-const#let%E3%81%A8const%E3%81%AE%E4%BD%BF%E3%81%84%E5%88%86%E3%81%91

takumi-333 replied:
そうなのか、ありがとう！！

DoiRyoto commented on components/top/Reviews.tsx:
良さそう

DoiRyoto commented on actions/review.action.ts:
### 183行目

これは分野があれば分野から選んできて、なければレビュー全部返す処理だと思うけど、コードだけだと分かりにくいので、適宜コメントで補足が欲しい。

DoiRyoto commented on lib/utils.ts:
resultはフラグで、1つでも分野が同じものがあればtrueになるなら、letを使わずにfor文の中でtrueを返せばよいのでは。

ループ抜けるならfalse返すだろうし。

takumi-333 committed:
[fix]コードレビューの指摘箇所を修正
・(home)/page.tsxでletを排除
・fetchReviewsByFields()メソッドにコメントを追加
・checkInStringArray()関数の実装修正

takumi-333 committed:
[仕様変更]
・トップページでtagが存在する時、tagによる検索のみに変更
